packages/odf/modals/s3-browser/delete-and-empty-bucket/EmptyBucketModal.tsx

```diff
@@ -0,0 +1,275 @@
+  const [inProgress, setInProgress] = React.useState(false);
+
+  const onEmpty = async (event) => {
+    event.preventDefault();
+    setInProgress(true);
+
+    try {
+      let isTruncated = true;
+      let keyMarker: string;
+      let versionIdMarker: string;
+
+      while (isTruncated) {
+        const deleteObjectKeys = [];
+        const searchParams = {
+          Bucket: bucketName,
+          KeyMarker: keyMarker,
+          VersionIdMarker: versionIdMarker,
+        };
+
+        // eslint-disable-next-line no-await-in-loop
+        const objects = await noobaaS3.listObjectVersions(searchParams);
+
+        if (objects?.Versions) {
+          deleteObjectKeys.push(
+            ...objects.Versions.map((object) => ({
+              Key: object.Key,
+              VersionId: object.VersionId,
+            }))
+          );
+        }
+        if (objects?.DeleteMarkers) {
+          deleteObjectKeys.push(
+            ...objects.DeleteMarkers.map((marker) => ({
+              Key: marker.Key,
+              VersionId: marker.VersionId,
+            }))
+          );
+        }
+
+        // eslint-disable-next-line no-await-in-loop
+        const deleteResponse = await noobaaS3.deleteObjects({
+          Bucket: bucketName,
+          Delete: { Objects: deleteObjectKeys },
+        });
+
+        setEmptyBucketResponse({
+          response: deleteResponse,
+          bucketName: bucketName,
+        });
+
+        isTruncated = objects.IsTruncated;
+        keyMarker = objects.NextKeyMarker;
+        versionIdMarker = objects.NextVersionIdMarker;
+      }
+
+      setInProgress(false);
+    } catch {
+      setInProgress(false);
+    } finally {
+      closeModal();
+      refreshTokens?.();
+    }
+  };
+
+  return (
+    <Modal
+      title={t('Empty bucket permanently?')}
+      titleIconVariant="warning"
+      isOpen={isOpen}
+      onClose={closeModal}
+      variant={ModalVariant.medium}
+      description={
+        <TextContent className="text-muted">
+          <Text component={TextVariants.p}>
+            {t(
+              'Emptying the bucket deletes all objects in the bucket and cannot be undone. Objects added to the bucket while the empty bucket action is in progress might be deleted.'
+            )}
+          </Text>
+          <Text component={TextVariants.p}>
+            {t(
+              'To prevent new objects from being added to this bucket while the empty bucket action is in progress, you might need to update your bucket policy to stop objects from being added to the bucket.'
+            )}
+          </Text>
+        </TextContent>
+      }
+      actions={[
+        <div>
+          {inProgress && (
+            <Text className="text-muted pf-v5-u-mb-sm">
+              <em>
+                {t(
+                  'The bucket is being emptied. This may take a while to empty'
+                )}
+              </em>
+            </Text>
+          )}
+          <div className="">
```

Comment:
do we need this div ??
```suggestion
          <div>
```